rapid7_insightidr/komand_rapid7_insightidr/actions/advanced_query_on_log/action.py

```diff
@@ -85,8 +67,14 @@ def get_results_from_callback(self, callback_url: str, timeout: int) -> [object]
         results_object = response.json()
         log_entries = results_object.get("events")
 
+
+        if results_object.get("links"):
+            callback_url = results_object.get("links")[0].get("href")
```

Comment:
Any chance "links" returns an empty list? If so, that will crash the plugin here. "href" could also return None, and None cannot be iterated over which will crash on line 77.

Reply:
No. If the links entry is there, it will have a links key.